tests/fixtures/liteclient/devicehsm.cc

```diff
@@ -0,0 +1,115 @@
+/**
+* The Device actions to register with the HSM token
+*
+*  1) Generates the ssl configuration file to allow using the HSM (softhsm)
+*  2) Stores the device key in the HSM (public/private keypair)
+*  3) Creates a CSR and requests its CRT from the RootCA
+*  4) Imports the CRT into the HSM.
+*  5) Generates HSM debug information if needed
+*
+*/
+class DeviceHsm {
+ public:
+  DeviceHsm(SoftHsm* hsm, RootCaPKI& rootCa, std::string conf) : hsm_(hsm), rootCa_(rootCa), cnf_(hsm_->path_ + conf) {
+    std::ofstream cnfOut(cnf_);
+    cnfOut << "openssl_conf = oc\n";
```

Comment:
Maybe it makes sense to use a string literal for the config template, e.g. https://github.com/foundriesio/aktualizr/blob/e2482e8426e328dac63e8abd7cc55bd68994e44c/src/libaktualizr/utilities/xml2json_test.cc#L50 and then use std::snprintf(). Not critical, just might improve readability as well as might be useful for manual testing, so we can just copy-paste the config template.